<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="shared roxygen params"><title>shared roxygen params — params • quartzbio.edp</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="shared roxygen params — params"><meta property="og:description" content="shared roxygen params"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">quartzbio.edp</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.00.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>shared roxygen params</h1>
      
      <div class="d-none name"><code>params.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>shared roxygen params</p>
    </div>


    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>account_id</dt>
<dd><p>an Account ID as a string.</p></dd>
<dt>alive</dt>
<dd><p>whether to select the Tasks that are alive, i.e. not finished
or failed.</p></dd>
<dt>all</dt>
<dd><p>whether to fetch all data, by iterating if needed.</p></dd>
<dt>ancestor_id</dt>
<dd><p>an object ID of an ancestor, for filtering.</p></dd>
<dt>as_data_frame</dt>
<dd><p>whether to convert the results as a data frame.</p></dd>
<dt>capacity</dt>
<dd><p>The dataset capacity level (small, medium, or large).</p></dd>
<dt>client_id</dt>
<dd><p>the client ID for the application.</p></dd>
<dt>commit_mode</dt>
<dd><p>There are four commit modes that can be selected depending on the scenario:
append (default), overwrite, upsert, and delete.</p></dd>
<dt>conn</dt>
<dd><p>a EDP connection object (as a named list or environment)</p></dd>
<dt>dataset_id</dt>
<dd><p>a Dataset ID as a string</p></dd>
<dt>data_type</dt>
<dd><p>the data type. one of:</p><ul><li><p>auto (the default)</p></li>
<li><p>boolean - Either True, False, or null</p></li>
<li><p>date - A string in ISO 8601 format, for example: <code>2017-03-29T14:52:01</code></p></li>
<li><p>double - A double-precision 64-bit IEEE 754 floating point.</p></li>
<li><p>float - single-precision 32-bit IEEE 754 floating point.</p></li>
<li><p>integer     A signed 32-bit integer with a minimum value of -231 and a maximum value of 231-1.</p></li>
<li><p>long     A signed 64-bit integer with a minimum value of -263 and a maximum value of 263-1.</p></li>
<li><p>object     A key/value, JSON-like object, similar to a Python dictionary.</p></li>
<li><p>string     A valid UTF-8 string up to 32,766 characters in length.</p></li>
<li><p>text     A valid UTF-8 string of any length, indexed for full-text search.</p></li>
<li><p>blob     A valid UTF-8 string of any length, not indexed for search.</p></li>
</ul></dd>
<dt>depth</dt>
<dd><p>the depth of the object in the Vault as an integer (0 means root)</p></dd>
<dt>description</dt>
<dd><p>the description as a string.</p></dd>
<dt>entity_type</dt>
<dd><p>A valid entity type:</p><ul><li><p>dataset - a Dataset ID (510110013133189334)</p></li>
<li><p>gene - A gene (EGFR)</p></li>
<li><p>genomic_region - A genomic region (GRCH38-7-55019017-55211628)</p></li>
<li><p>literature - A PubMed ID (19915526)</p></li>
<li><p>sample - A sample identifier (TCGA-02-0001)</p></li>
<li><p>variant - A genomic variant (GRCH38-7-55181378-55181378-T)</p></li>
</ul></dd>
<dt>env</dt>
<dd><p>Custom client environment.</p></dd>
<dt>exclude_fields</dt>
<dd><p>A list of fields to exclude in the results, as a character vector.</p></dd>
<dt>exclude_group_id</dt>
<dd><p>a group ID to exclude.</p></dd>
<dt>expression</dt>
<dd><p>EDP xpressions are Python-like formulas that can be used to pull
data from datasets, calculate statistics, or run advanced algorithms.</p></dd>
<dt>facets</dt>
<dd><p>A valid facets objects.</p></dd>
<dt>fields</dt>
<dd><p>The fields to add.</p></dd>
<dt>field_id</dt>
<dd><p>a Field object ID.</p></dd>
<dt>file_id</dt>
<dd><p>a file Object ID.</p></dd>
<dt>filename</dt>
<dd><p>an Object filename, without the parent folder (e.g. "foo.csv")</p></dd>
<dt>filters</dt>
<dd><p>a filter expression as a JSON string.</p></dd>
<dt>full_path</dt>
<dd><p>an Object full path, including the account, vault and path.</p></dd>
<dt>glob</dt>
<dd><p>a glob (full path with wildcard characters) which searches
objects for matching paths (case-insensitive).</p></dd>
<dt>include_errors</dt>
<dd><p>whether to include errors in the output.</p></dd>
<dt>is_list</dt>
<dd><p>whether the result is expected to be a list.</p></dd>
<dt>limit</dt>
<dd><p>The maximum number of elements to fetch, as an integer.
See also <code>page</code>.</p></dd>
<dt>local_path</dt>
<dd><p>the path of a local file.</p></dd>
<dt>md5</dt>
<dd><p>a MD5 fingerprint, as a string.</p></dd>
<dt>metadata</dt>
<dd><p>metadata as a named list.</p></dd>
<dt>mimetype</dt>
<dd><p>the MIME type of the Object.</p></dd>
<dt>min_distance</dt>
<dd><p>used in conjuction with the ancestor_id filter to only include
objects at a minimum distance from the ancestor.</p></dd>
<dt>object_type</dt>
<dd><p>the type of an object, one of  "file", "folder", or "dataset".</p></dd>
<dt>ordering</dt>
<dd><p>A list of fields to order/sort the results with, as a character vector.</p></dd>
<dt>offset</dt>
<dd><p>the file offset (starts from 0).</p></dd>
<dt>page</dt>
<dd><p>The number of the page to fetch, as an integer. starts from 1.
See also <code>limit</code>.</p></dd>
<dt>path</dt>
<dd><p>the path of an object, with the folders (e.g. "/d1/d2/foo.csv").</p></dd>
<dt>parent_object_id</dt>
<dd><p>the ID of the parent of the Object.</p></dd>
<dt>query</dt>
<dd><p>a string that matches any objects whose path contains that string.</p></dd>
<dt>raw</dt>
<dd><p>whether to return the raw response.</p></dd>
<dt>records</dt>
<dd><p>The data to annotate as a data frame.</p></dd>
<dt>regex</dt>
<dd><p>A regular expression, as a string, to filter the results with.</p></dd>
<dt>size</dt>
<dd><p>The size of the object.</p></dd>
<dt>status</dt>
<dd><p>a Task status, one of (running, queued, pending, completed, failed
)</p></dd>
<dt>storage_class</dt>
<dd><p>The Storage class of the vault <code>('Standard', 'Standard-IA', 'Essential', 'Temporary', 'Performance', 'Archive')</code> as a string.</p></dd>
<dt>sync</dt>
<dd><p>whether to proceed in synchronous mode, i.e to wait for all
sub tasks to finish before returning.</p></dd>
<dt>tag</dt>
<dd><p>a single tag as a string.</p></dd>
<dt>tags</dt>
<dd><p>a list of tags as a character vector.</p></dd>
<dt>target_fields</dt>
<dd><p>A list of valid dataset fields to create or override in the import,
as a character vector.</p></dd>
<dt>task_id</dt>
<dd><p>an (ECS) Task ID as a string.</p></dd>
<dt>url_template</dt>
<dd><p>A URL template with one or more "value" sections that will be
interpolated with the field value and displayed as a link
in the dataset table.</p></dd>
<dt>user_id</dt>
<dd><p>a user id (or User object) as a string</p></dd>
<dt>vault_id</dt>
<dd><p>a Vault ID as a string (e.g. "19").</p></dd>
<dt>vault_name</dt>
<dd><p>a Vault name as a string (e.g. "Public").</p></dd>
<dt>vault_full_path</dt>
<dd><p>a Vault full path, as a string (e.g. "quartzbio:Public")</p></dd>
<dt>vault_path</dt>
<dd><p>a Vault path, as a string (e.g. "/d1/d2/foo.csv")</p></dd>
<dt>vault_type</dt>
<dd><p>the type of vault ('user', 'general') as a string</p></dd>
<dt>...</dt>
<dd><p>Additional query parameters, passed to .request().</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p><a href="https://docs.solvebio.com/" class="external-link">https://docs.solvebio.com/</a></p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Karl Forner.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.1.</p>
</div>

    </footer></div>

  

  

  </body></html>

